### PR TITLE
inspect all and cpu profile

### PR DESCRIPTION
**Pull request by rebornix**
This PR includes 

* `--inspect-all` in CLI. Runs Main/Renderer/Terminal/Search processes all in debug mode.
* `--cpu-profile` in CLI. Profile against a port.
* `--status` now prints debug port
* A workbench command which reloads VSCode in `--inspect-all` mode.

This PR requires https://github.com/jrieken/v8-inspect-profiler/pull/1 to work.

### CODE REVIEW

File: src/typings/v8-inspect-profiler.d.ts [hunk 1]
```diff
@@ -31,7 +31,18 @@ declare module 'v8-inspect-profiler' {
 		stop(afterDelay?: number): PromiseLike<ProfileResult>;
```
Comment (rebornix) [inline src/typings/v8-inspect-profiler.d.ts]: it's merged.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: thanks.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: update.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: get call back from extension.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: new comment.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: 123

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: 345

Comment (kieferrm) [reply src/typings/v8-inspect-profiler.d.ts]: sdadasdasd

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: now I know Kai made some comments successfully

Comment (kieferrm) [reply src/typings/v8-inspect-profiler.d.ts]: Indeed

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: great

File: src/typings/v8-inspect-profiler.d.ts [hunk 2]
```diff
@@ -31,7 +31,18 @@ declare module 'v8-inspect-profiler' {
...
 	}
 
-	export function startProfiling(options: { port: number, tries?: number, retyWait?: number }): PromiseLike<ProfilingSession>;
+	export interface Target {
+		description: string,
+    	devtoolsFrontendUrl: string,
+		id: string,
```
Comment (rebornix) [inline src/typings/v8-inspect-profiler.d.ts]: line 37

File: src/vs/base/node/ports.ts [hunk 3]
```diff
@@ -32,6 +32,11 @@ export function findFreePort(startPort: number, giveUpAfter: number, timeout: nu
 	});
 }
 
+export function findRandomFreePort(): Thenable<number> {
+	const port = Math.floor(Math.random() * (20000 - 9000) + 9000);
```
Comment (rebornix) [inline src/vs/base/node/ports.ts]: lol

File: src/vs/base/node/ports.ts [hunk 4]
```diff
@@ -32,6 +32,11 @@ export function findFreePort(startPort: number, giveUpAfter: number, timeout: nu
...
+	return findFreePort(port, 10, 5000);
+}
+
```
Comment (rebornix) [inline src/vs/base/node/ports.ts]: cool

Comment (rebornix) [reply src/vs/base/node/ports.ts]: 不错

File: src/typings/v8-inspect-profiler.d.ts (hunk 1 above)
Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: *italic*

File: src/typings/v8-inspect-profiler.d.ts (hunk 2 above)
Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: lol

File: src/typings/v8-inspect-profiler.d.ts (hunk 1 above)
Comment (RMacfarlane) [reply src/typings/v8-inspect-profiler.d.ts]: asdf

File: src/typings/v8-inspect-profiler.d.ts (hunk 2 above)
Comment (RMacfarlane) [reply src/typings/v8-inspect-profiler.d.ts]: asdf

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: fdsa

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: I made some refactoring.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: testing.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: cool2

File: src/typings/v8-inspect-profiler.d.ts (hunk 1 above)
Comment (kieferrm) [reply src/typings/v8-inspect-profiler.d.ts]: **bold** _italic_

Comment (RMacfarlane) [reply src/typings/v8-inspect-profiler.d.ts]: cool

File: src/vs/base/node/ports.ts [hunk 5]
```diff
@@ -32,6 +32,11 @@ export function findFreePort(startPort: number, giveUpAfter: number, timeout: nu
 	});
```
Comment (rebornix) [inline src/vs/base/node/ports.ts]: 33

File: src/typings/v8-inspect-profiler.d.ts (hunk 1 above)
Comment (kieferrm) [reply src/typings/v8-inspect-profiler.d.ts]: **bold** _italic_

File: src/vs/base/node/ports.ts [hunk 6]
```diff
@@ -32,6 +32,11 @@ export function findFreePort(startPort: number, giveUpAfter: number, timeout: nu
...
 }
```
Comment (rebornix) [inline src/vs/base/node/ports.ts]: after 34.

File: src/vs/code/electron-main/diagnostics.ts [hunk 7]
```diff
@@ -162,18 +165,33 @@ function formatProcessItem(mapPidToWindowTitle: Map<number, string>, output: str
 	let name: string;
 	if (isRoot) {
 		name = `${product.applicationName} main`;
+		let matches = RENDERER_DEBUG_PATTERN.exec(item.cmd);
+		if (matches && matches.length === 2) {
+			rendererDebugPort = matches[1];
+		}
 	} else {
```
Comment (RMacfarlane) [inline src/vs/code/electron-main/diagnostics.ts]: comment range

File: src/typings/v8-inspect-profiler.d.ts [hunk 8]
```diff
@@ -31,7 +31,18 @@ declare module 'v8-inspect-profiler' {
...
+		title: string,
```
Comment (rebornix) [inline src/typings/v8-inspect-profiler.d.ts]: this is a new comment.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: reply.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: incremental update.

File: src/typings/v8-inspect-profiler.d.ts [hunk 9]
```diff
@@ -31,7 +31,18 @@ declare module 'v8-inspect-profiler' {
...
+		type: string,
+		url: string,
+		webSocketDebuggerUrl: string
+	}
```
Comment (rebornix) [inline src/typings/v8-inspect-profiler.d.ts]: list tabs.

File: src/vs/base/node/ports.ts (hunk 4 above)
Comment (rebornix) [reply src/vs/base/node/ports.ts]: 点赞

Comment (rebornix) [reply src/vs/base/node/ports.ts]: revert

File: src/typings/v8-inspect-profiler.d.ts (hunk 9 above)
Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: no command any more.

File: src/typings/v8-inspect-profiler.d.ts (hunk 2 above)
Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: rerender textarea after post message.

Comment (rebornix) [reply src/typings/v8-inspect-profiler.d.ts]: we need to update comment form css after post message.

**Comment by rebornix** [comment]
test UI update.

File: src/vs/code/electron-main/diagnostics.ts [hunk 10]
```diff
@@ -17,6 +17,9 @@ import { isWindows } from 'vs/base/common/platform';
 import { app } from 'electron';
 import { basename } from 'path';
 
+const DEBUG_FLAGS_PATTERN = /\s--(inspect|debug)(-brk)?(=([0-9]+))/;
+const RENDERER_DEBUG_PATTERN = /--remote-debugging-port=([0-9]+)/;
```
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/vs/code/electron-main/diagnostics.ts]: comment

**Comment by rebornix** [comment]
Close it now as we are not likely to have it merged.

**Pull request closed without merging**